worker/uniter/operation/runhook.go

```diff
@@ -153,7 +153,16 @@ func (rh *runHook) beforeHook(state State) error {
 			Status: string(status.Maintenance),
 			Info:   "cleaning up prior to charm deletion",
 		})
+	case hooks.LeaderElected:
+		// Check if leadership has changed between queueing of the hook and
+		// Actual execution. Set the error if we are no longer the leader.
+		isLeader := false
+		isLeader, err = rh.runner.Context().IsLeader()
+		if err == nil && !isLeader {
+			err = context.ErrIsNotLeader
```

Comment:
I think that if the unit is no longer the leader, we don't want to leave the uniter in an error state.

Perhaps just log an info message saying skipping due leader-elected to no longer leader, and return nil?

Reply:
@howbazaar, @wallyworld 

Can we return directly with a nil state, or should we behave as if the hook had run despite not calling `RunHook`? I.e do we:
- call `callbacks.NotifyHookCompleted`?
- call `afterHook`?
- return a changed state?

Apologies for the churn; first time in the uniter.

Reply:
@manadart 
The correct approach is to modify the runhook Prepare() method to return ErrSkipExecute if the unit is no longer a leader. So take the logic you have here and move to Prepare(). That should be all you need to do

Reply:
@wallyworld Suggested change is applied.